### PR TITLE
feat(CX-2943): add select artist modal to median sale price

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **feature**

This PR solves [CX-2943]

### Description

This PR adds the select artist screen as a **modal**. The reason behind that is as discussed with Pierluca and Dascha the ease of the implementation and to use the existing `ModalDialog`.

https://user-images.githubusercontent.com/11945712/197208057-c47fb121-db30-4c83-8cfa-f8942eddca35.mov

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CX-2943]: https://artsyproduct.atlassian.net/browse/CX-2943?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/MyCollectionInsights/myCollectionInsightsRoutes.tsx [hunk 1]
```diff
@@ -16,6 +17,19 @@ export const myCollectionInsightsRoutes: AppRouteConfig[] = [
   {
     path: "/my-collection/median-sale-price-at-auction/:artistID",
     getComponent: () => MyCollectionInsightsMedianSalePriceAtAuction,
+    onClientSideRender: () => {
+      MyCollectionInsightsMedianSalePriceAtAuction.preload()
+    },
+    query: graphql`
```
Comment (MrSltun) [inline src/Apps/MyCollectionInsights/myCollectionInsightsRoutes.tsx:23]: I believe the query is missing the variables from the URL, it needs a `prepareVariables` field similar to [this one](https://github.com/artsy/force/blob/main/src/Apps/Artwork/artworkRoutes.tsx#L19)

File: src/Apps/MyCollectionInsights/Routes/MyCollectionInsightsMedianSalePriceAtAuction/MyCollectionInsightsMedianSalePriceAtAuction.tsx [hunk 2]
```diff
@@ -9,6 +23,44 @@ export const MyCollectionInsightsMedianSalePriceAtAuction = () => {
...
+          onClick={() => setShowSelectArtistModal(!showSelectArtistModal)}
+        >
+          Change Artist
+        </Clickable>
+      </Flex>
+
+      {showSelectArtistModal && (
+        <ModalDialog
+          onClose={() => setShowSelectArtistModal(false)}
+          width={"100%"}
```
Comment (MrSltun) [inline src/Apps/MyCollectionInsights/Routes/MyCollectionInsightsMedianSalePriceAtAuction/MyCollectionInsightsMedianSalePriceAtAuction.tsx:53]: **Suggestion:** Maybe we should limit the width on desktop view? let's say 800px? and 100% on mWeb?

```suggestion
          width={["100%", 800]}
```

**Review by MrSltun — APPROVED** [review]
Looks Good! ✨ I just left some comments on it

File: src/Apps/MyCollectionInsights/Routes/MyCollectionInsightsMedianSalePriceAtAuction/MyCollectionInsightsMedianSalePriceAtAuction.tsx (hunk 2 above)
Comment (MounirDhahri) [reply src/Apps/MyCollectionInsights/Routes/MyCollectionInsightsMedianSalePriceAtAuction/MyCollectionInsightsMedianSalePriceAtAuction.tsx:53]: good idea, I will do it and follow up later with Dascha

File: src/Apps/MyCollectionInsights/myCollectionInsightsRoutes.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/Apps/MyCollectionInsights/myCollectionInsightsRoutes.tsx:23]: this is a good comment, the url params seem to be automatically injected while respecting the format they're defined as. You can see in the video below how they're being injected properly without having the `prepareVariables` method defined. `prepareVariable` is working as middleware and it wasn't needed here. We might need it later if we want read the medium at a further point

**Review by MrSltun — APPROVED** [review]
Okay great! I'll merge this now and we can go back to review the design and experience

File: src/Apps/MyCollectionInsights/myCollectionInsightsRoutes.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/Apps/MyCollectionInsights/myCollectionInsightsRoutes.tsx:23]: I forgot to link this video, as you see, params are getting injected here without the middleware

https://user-images.githubusercontent.com/11945712/197558099-cfe9f175-58f4-43bc-bc40-e6e250b0e373.mov

**Pull request merged**